python/social_bot/tasks.py

```diff
@@ -1137,3 +1142,249 @@ def task_specific_observation(self, agent):
              palm_pos]).flatten()
         return np.concatenate(
             (obs, finger_contacts, agent_pose, joints_states), axis=0)
+
+
+@gin.configurable
+class Stack(Task):
+    """
+    A task to stack several wood cubes together.
+    """
+
+    compatible_agents = ['youbot_noplugin']
+
+    def __init__(self,
+                 env,
+                 max_steps,
+                 max_distance=0.5,
+                 min_distance=0.3,
+                 objects_num=3,
+                 objects_random_range=0.25,
+                 object_half_size=0.025,
+                 success_distance_thresh=0.03,
+                 reward_shaping=True,
+                 reward_weight=1.0):
+        """
+        Args:
+            env (gym.Env): an instance of Environment
+            max_steps (int): episode will end if not complet the task in so many steps, recommend to be 150
+                for agent youbot_noplugin and object 5cm cube
+            max_distance (float): the max distance from the random object's location to the agent
+            min_distance (float): the min distance from the random object's location to the agent
+            objects_num (int): the number of objects to stacking
+            objects_random_range (float): the range of objects distributed
+            object_half_size (float): Note that model for staking task should be of no offset inside the model.
+                This means an initial pose of 0 height makes half of the obejct underground. This specify the
+                initial height of object's center, e.g, half the edge length of a cube, or radius of a ball.
+            success_distance_thresh (float): consider success if the objects' x-y plance distance is within
+                this threshold
+            reward_shaping (bool): if false, the reward is -1/0/1, otherwise the 0 case will be replaced
+                with shapped reward.
+            reward_weight (float): the weight of the reward
+        """
+        super().__init__(
+            env=env, max_steps=max_steps, reward_weight=reward_weight)
+        assert self._agent.type in self.compatible_agents, "Stack task only support youbot_noplugin for now"
+        self._reward_shaping = reward_shaping
+        self._palm_link = 'youbot_noplugin::gripper_palm_link'
+        self._finger_link_l = 'youbot_noplugin::gripper_finger_link_l'
+        self._finger_link_r = 'youbot_noplugin::gripper_finger_link_r'
+        self._object_collision_name = '::wood_cube_5cm_without_offset::link::collision'
+        self._object_half_size = object_half_size
```

Comment:
It can be get from gazebo when the model has a default internal offset. To due with the situation when the cube is rotated conveniently, the internal offset has been removed so we can not get that now. 
This parameter is needed to insert/move the model (or its half will be below ground) and also useful in reward shaping.

Reply:
Have you tried using the CollisionBoundingBox of a gazebo Model? https://osrf-distributions.s3.amazonaws.com/gazebo/api/dev/classgazebo_1_1physics_1_1Entity.html#a2f67b75d71f09ace9a15a3dcccb812dc